### PR TITLE
Async room id

### PR DESCRIPTION
**Pull request by hiroshihorie**

### CODE REVIEW

File: Sources/LiveKit/Core/Room.swift [hunk 1]
```diff
@@ -28,11 +28,19 @@ public class Room: NSObject, ObservableObject, Loggable {
 
     // MARK: - Public
 
-    @objc
+    @objc(sid)
     /// Server assigned id of the Room.
     /// This will be empty until ``RoomDelegate/room(_:didUpdateRoomId:)`` is called.
```
Comment (davidzhao) [inline Sources/LiveKit/Core/Room.swift:33]: this comment isn't needed anymore?

File: Sources/LiveKit/Core/Room.swift [hunk 2]
```diff
@@ -28,11 +28,19 @@ public class Room: NSObject, ObservableObject, Loggable {
...
     public var sid: Sid? { _state.sid }
 
+    // Work-around error: Property with 'throws' or 'async' is not representable in Objective-C
+    /// Server assigned id of the Room. async version of ``Room/sidValue``.
+    public var asyncSid: Sid {
```
Comment (davidzhao) [inline Sources/LiveKit/Core/Room.swift:38]: how would one use this value? `await room.asyncSid()` ? should we state usage in the comment?

Comment (hiroshihorie) [reply Sources/LiveKit/Core/Room.swift:38]: Xcode will show it's an await-able property.

**Comment by hiroshihorie** [comment]
After merging, I realized property and method can have same name so I've made it to a method (additionally adds ObjC support) `await room.sid()` 
https://github.com/livekit/client-sdk-swift/commit/7ae1aa5517f269e6743713816cf7d52f902ca041

**Pull request merged**